Add @examples trait conversion to Smithy2OpenAPI converter for REST protocols. (#1284)
awslabs/smithy: docs/source/1.0/guides/converting-to-openapi.rst

        ]
    }

> ----------------

Reviewer: needs to span the entire lenght of the title. So does line 809

Author: resolved.

``@examples`` trait conversion
----------------

In Smithy, example values of input structure members and the corresponding
output or error structure members for an operation are grouped together
into one set of example values for an operation. Below is an example "unit" of ``FooOperation``
operation shape, which shows this logical grouping.

.. code-block:: smithy

    apply FooOperation @examples(
        [
            {
                title: "valid example",
                documentation: "valid example doc",
                input: {
                    bar: "1234"
                },
                output: {
                    baz: "5678"
                },
            }
        ]
    )

> However, example values in OpenAPI are scattered throughout the model, with each example value

Reviewer: Oh interesting. It's unfortunate that you can't provide a fully example of a request's input (headers + body) or response headers + body. You have to break them up into isolated examples that don't show full messages.

contained by the OpenAPI object the example value is for.

> For example, below is an example output of above Smithy example value.

Reviewer: ```suggestion
The following is an example OpenAPI model for the above Smithy example value.
```

Author: resolved.
